fendermint/app/src/store.rs

```diff
@@ -37,3 +42,75 @@ where
         fvm_ipld_encoding::from_slice(repr).map_err(|e| KVError::Codec(Box::new(e)))
     }
 }
+
+/// A `Blockstore` and `BitswapStore` implementation we can pass to the IPLD Resolver.
+pub struct BitswapBlockstore {
+    /// The `Blockstore` implementation where we the FVM actors store their data.
+    ///
+    /// This must not be written to by Bitswap operations, because that could result
+    /// in some nodes having some data that others don't, which would lead to a
+    /// consensu failure. We can use read data from it, but not write to it.
+    state_store: NamespaceBlockstore,
+    /// The `Blockstore` implementation where Bitswap operations can write to.
+    bit_store: NamespaceBlockstore,
```

Comment:
I really like this dual blockstore approach where we have the `bit_store` as cache and we expose the `state_store`. Really cool!

Reply:
Yeah we have to be careful that we don't put stuff into the `state_store` that isn't a result of actor execution. 

In order to make cross-message data available to Bitswap I think we'll have to shovel it form Solidity to some IPLD-native data, and we can store that in the `state_store` because it happens automatically as part of execution. We could store it in `bit_store` too. 

What we receive must go into the `bit_store` though, and then when we want to execute, we have to turn them back to Solidity by pulling from `bit_store` or `state_store`. 

I imagine using namespaces makes stuff slower but safety first :police_officer: